Creating agenda

navro opened the pull request.

davidwessman commented on app/assets/stylesheets/partials/_agenda.scss:
```diff
@@ -0,0 +1,11 @@
+      td,
+      &.even td,
+      &.odd td {
+        background-color: rgba(252, 152, 0, 0.5);
```
Perhaps we should add `$orange-highlight = rgba(252, 152, 0, 0.5);` to 
https://github.com/fsek/voting/blob/master/app/assets/stylesheets/colors.scss

davidwessman commented on app/models/agenda.rb:
```diff
@@ -0,0 +1,69 @@
+
+  private
+
+  def parent_validation
```
Would like some test for this, but it seems valid 😄

davidwessman commented on app/views/admin/adjustments/_form.html.erb:
```diff
@@ -0,0 +1,10 @@
+  <%= f.input :presence, include_blank: false, collection: adjustment_collection %>
+  <%= f.hidden_field :user_id, value: @adjustment.user_id %>
+  <div class="table">
+    <%= f.button :submit, class: 'btn primary' %>
```
`class: 'btn primary'` is not needed here. Added through `SimpleForm` setup.

davidwessman commented on app/views/admin/adjustments/edit.html.erb:
```diff
@@ -0,0 +1,8 @@
+    <h1><%= t('adjustment.edit') %></h1>
+  </div>
+  <%= render 'form' %>
+    <%= link_to t(:destroy_attr, attr: model_name(Adjustment)),
```
I tried to use these translations and I think it ended up rather messy.
I do prefer to use `t('adjustment.destroy')`.

davidwessman commented on app/views/layouts/_admin_dropdown.html.erb:
```diff
@@ -48,5 +48,10 @@
         <%= link_to t('user.vote_administration'), admin_vote_users_path %>
       </li>
     <% end %>
+    <% if can? :manage, Agenda %>
```
`:modify`

davidwessman commented:
Gonna test it out locally as well, looking really good 😄

davidwessman commented:
The buttons on the admin pages are great, would be good to have one for the Agenda as well 😄 
Other than my comments, +1

navro commented:
> The buttons on the admin pages are great, would be good to have one for the Agenda as well :smile:
> Other than my comments, +1

I fixed all your comments except the authorizations. Let's wait until we decide what to do in `web`. I also added some tests for `Agenda` :smile:

davidwessman commented on app/controllers/admin/adjustments_controller.rb:
```diff
@@ -0,0 +1,49 @@
+class Admin::AdjustmentsController < ApplicationController
+  load_permissions_and_authorize_resource
```
This controller is only interacted by admins, so we could remove the `before_action`

davidwessman commented:
:+1: 
I like the admin namespace.
If we have a *normal* controller as well I think we should stick with `authorize!(:manage, Model)` until we create an `AdminAbility`. 

I think we should get this merged (since we want to connect a lot to it) and then fix the authorization through the whole project later.

navro commented:
> :+1:
> I like the admin namespace.
> If we have a normal controller as well I think we should stick with authorize!(:manage, Model) until we create an AdminAbility.
> 
> I think we should get this merged (since we want to connect a lot to it) and then fix the authorization through the whole project later.

I agree :)
